components/sup/src/manager/service/spec.rs

```diff
@@ -249,7 +254,7 @@ impl ServiceSpec {
     /// * If any given service binds are in neither required nor optional package binds
     fn validate_binds(&self, package: &PackageInstall) -> Result<()> {
         let mut svc_binds: HashSet<String> =
-            HashSet::from_iter(self.binds.iter().cloned().map(|b| b.name));
+            HashSet::from_iter(self.binds.iter().map(|b| b.name().to_string()));
```

Comment:
While it is a bit longer in this case, I prefer this form and it obviates the need for creation of closures:
```suggestion
            HashSet::from_iter(self.binds.iter().map(ServiceBind::name).map(str::to_string));
```
Perhaps the compiler can optimize this away, but I alway like to hand it pre-existing functions when they exist to do the job.

Reply:
Additionally, you might be able to get away with:
```suggestion
            HashSet::from_iter(self.binds.iter().map(ServiceBind::name));
```
Making svc_binds a `HashSet<&str>`. It requires other changes, but avoids unnecessary cloning. I haven't worked it out completely, but I think it should be possible.

Reply:
Yeah, this made the code better.  I also added some tests around this as the logic was all untested.